src/toast/Toast.tsx

```diff
@@ -0,0 +1,163 @@
+        const pos = options.position === undefined ? Direction.TOP_RIGHT : options.position;
+
+        //As this is a simple toast, allowing one toast to be shown at a time
+        if (this.state.isToastRendering === false) {
+            //set the state to show the toast and setTimeout to hide the toast
+            this.setState(
+                {
+                    isToastRendering: true,
+                    msg: message,
+                    type: toastType,
+                    position: pos,
+                    hideAfter: hideTime,
+                },
+                () => {
+                    setTimeout(
+                        () =>
+                            this.setState({
+                                isToastRendering: false,
+                                msg: EMPTY_MESSAGE,
+                                type: MessageType.SUCCESS,
+                                position: Direction.TOP_RIGHT,
+                                hideAfter: DEFAULT_HIDE_TIME,
+                            }),
+                        hideTime,
+                    );
+                },
+            );
+        }
+    };
+
+    render() {
+        let classes = "";
+        let classname = "";
+        if (this.state.isToastRendering === true) {
+            //create styles based on the position given
+            classes = TOAST_CLASSES.VISIBLE + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.BOTTOM)
+                ? TOAST_CLASSES.BOTTOM + " " + TOAST_CLASSES.MARGIN_BOTTOM
+                : TOAST_CLASSES.TOP + " " + TOAST_CLASSES.MARGIN_TOP;
+            classes = classes + " " + classname + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.RIGHT)
+                ? TOAST_CLASSES.RIGHT + " " + TOAST_CLASSES.MARGIN_RIGHT
+                : TOAST_CLASSES.LEFT + " " + TOAST_CLASSES.MARGIN_LEFT;
+            classes = classes + " " + classname;
+        } else {
+            classes = TOAST_CLASSES.INVISIBLE + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.BOTTOM)
+                ? TOAST_CLASSES.NEG_MARGIN_BOTTOM
+                : TOAST_CLASSES.NEG_MARGIN_TOP;
+            classes = classes + " " + classname + " ";
+            classname = (this.state.position || Direction.TOP_RIGHT).includes(Direction.RIGHT)
+                ? TOAST_CLASSES.NEG_MARGIN_RIGHT
+                : TOAST_CLASSES.NEG_MARGIN_LEFT;
+            classes = classes + " " + classname;
+        }
+
+        let classnames = `toast ${this.state.type} ${classes}`;
+        return (
+            <div className={classnames}>
+                <span className={PADDING_RIGHT}>
+                    <Icon shape={this.getIcon()} />
+                </span>
+                {this.state.msg}
+            </div>
+        );
+    }
+    /* Retrieve relevant icon based on type of message */
+    getIcon() {
```

Comment:
Can remove this if we just pass icon as a property that gets passed as the shape.

Reply:
Updated code